### PR TITLE
Fix 556/repo conversion times out

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Added support for splitting conversion into multiple parts.
bible conversions are split into separate conversions per book.

### CODE REVIEW

File: client/client_callback.py [hunk 1]
```diff
@@ -22,16 +24,30 @@ def __init__(self, job_data=None, cdn_bucket=None, gogs_url=None):
...
+        if not multipleProject:
+            owner_name, repo_name, commit_id = parts
+        else:
+            owner_name, repo_name, commit_id, part_count, part_id = parts
+            self.logger.debug('Multiple project, part {0} of {1}'.format(part_id,part_count))
+
         # The identifier is how to know which username/repo/commit this callback goes to
         s3_commit_key = 'u/{0}/{1}/{2}'.format(owner_name, repo_name, commit_id)
 
+        self.logger.debug('Callback for comming {0}...'.format(s3_commit_key))
```
Comment (phillip-hopper) [inline client/client_callback.py]: Is `comming` a typo?

File: client/client_callback.py [hunk 2]
```diff
@@ -42,32 +58,163 @@ def process_callback(self):
...
+            build_log_json['log'] = []
+        if not 'warnings' in build_log_json:
+            build_log_json['warnings'] = []
+        if not 'errors' in build_log_json:
+            build_log_json['errors'] = []
+
+    def cdnDownloadFile(self, cdn_handler, s3_part_key, filePath):
+        cdn_handler.download_file(s3_part_key, filePath)
+
+    def downloadFile(self, converted_zip_file, converted_zip_url):
```
Comment (phillip-hopper) [inline client/client_callback.py]: What is the purpose of this function? It appears to just call another function.

File: client/client_callback.py [hunk 3]
```diff
@@ -42,32 +58,163 @@ def process_callback(self):
...
+    def buildLogSanityCheck(self, build_log_json):
+        # sanity check
+        if not 'log' in build_log_json:
+            build_log_json['log'] = []
+        if not 'warnings' in build_log_json:
+            build_log_json['warnings'] = []
+        if not 'errors' in build_log_json:
+            build_log_json['errors'] = []
+
+    def cdnDownloadFile(self, cdn_handler, s3_part_key, filePath):
```
Comment (phillip-hopper) [inline client/client_callback.py]: What is the purpose of this function? It appears to just call another function.

File: client/client_callback.py [hunk 4]
```diff
@@ -42,32 +58,163 @@ def process_callback(self):
...
+        cdn_handler.download_file(s3_part_key, filePath)
+
+    def downloadFile(self, converted_zip_file, converted_zip_url):
+        download_file(converted_zip_url, converted_zip_file)
+
+    def getFinishedParts(self, cdn_handler, s3_commit_key):
```
Comment (phillip-hopper) [inline client/client_callback.py]: What is the purpose of this function? It appears to just call another function.

File: client/client_callback.py [hunk 5]
```diff
@@ -91,12 +238,21 @@ def process_callback(self):
...
-        project_file = os.path.join(tempfile.gettempdir(), 'project.json')
+        project_file = os.path.join(self.tempDir, 'project.json')
         write_file(project_file, project_json)
-        cdn_handler.upload_file(project_file, project_json_key, 0)
+        self.cdnUploadFile(cdn_handler, project_file, project_json_key, 0)
+        return project_json
 
-        # Now download the existing build_log.json file, update it and upload it back to S3
-        build_log_json = cdn_handler.get_json(s3_commit_key + '/build_log.json')
+    def cdnGetJsonFile(self, cdn_handler, project_json_key):
```
Comment (phillip-hopper) [inline client/client_callback.py]: What is the purpose of this function? It appears to just call another function.

File: tests/manager_tests/test_manager.py [hunk 6]
```diff
@@ -206,6 +206,22 @@ def tearDownClass(cls):
         for patch in ManagerTest.patches:
             patch.stop()
 
+    def test_child_job(self):
+        tx_manager = TxManager(**self.tx_manager_env_vars)
+        expectedCount = "4"
```
Comment (phillip-hopper) [inline tests/manager_tests/test_manager.py]: Is there a reason we are expecting integers to be cast as strings?

**Review by phillip-hopper — CHANGES_REQUESTED** [review]

File: client/client_callback.py [hunk 7]
```diff
@@ -22,16 +24,31 @@ def __init__(self, job_data=None, cdn_bucket=None, gogs_url=None):
         self.job = TxJob(job_data)
         self.cdn_bucket = cdn_bucket
         self.gogs_url = gogs_url
+        self.tempDir = tempfile.mkdtemp(suffix="", prefix="client_callback_")
```
Comment (richmahn) [inline client/client_callback.py]: As discussed on Slack, it is helpful for other programmers to at least have all variable names as lowercase with underscore, like the variables above this one. Should be like this in all the code for this PR.

**Review by richmahn — CHANGES_REQUESTED** [review]
Would be good to standardize variable names, using lowercase and underscores.

**Review by phillip-hopper — APPROVED** [review]

File: client/client_callback.py (hunk 7 above)
Comment (PhotoNomad0) [reply client/client_callback.py]: @rich - OK Changed color settings in IntelliJ so that I will notice the warnings.  Made a ton of fixes.

**Review by richmahn — APPROVED** [review]

**Review by richmahn — APPROVED** [review]
LGTM 👍

**Pull request merged**